### ISSUE TITLE
Serialise job-graphs onto the database

### ISSUE THREAD

**Issue by daemonblade** [OCA/queue#502]
**Is your feature request related to a problem?**

When chaining groups of large jobs (each group containing 1000's of jobs), Odoo fails with _MemoryError_

**Describe the solution you'd like**

If I understand the code correctly, DelayableGraph, DelayableChain and DelayableGraph are currently instantiated in-memory. Would it be feasible to re-implement these using ORM tables to achieve the same result? It would be slower, but scalable to large graphs.

**Comment by hparfr** [comment]
Can you postpone some jobs creation inside a job ? 

```
def generate_jobs(limit):
  while i < limit or i < CONST:
      delay().do_something()
  if i < limit:
      delay().generate_job(limit - i)
```

**Comment by daemonblade** [comment]
I'm currently attempting to port a project from another deferred job-queue system onto `queue_job`. The other project had a simple dependency system, which blocked on groups of jobs instead of single jobs. Each job was assigned a (plaintext) job-type, and an optional list of blocking job-types. Jobs could not start until all blocking job-types were completed.

It would possible to spend more time to decompose jobs so that a job could generate job on the queue; but some chains have no obvious direct depedencies.
